planning/autoware_path_generator/src/utils.cpp

```diff
@@ -247,5 +247,48 @@ std::vector<geometry_msgs::msg::Point> get_path_bound(
 
   return path_bound;
 }
+
+TurnIndicatorsCommand get_turn_signal(
+  const PathWithLaneId & path, const lanelet::LaneletMap & lanelet_map,
+  const geometry_msgs::msg::Pose & current_pose, const double current_vel,
+  const double search_distance, const double search_time, const double base_link_to_front)
+{
+  TurnIndicatorsCommand turn_signal;
+  turn_signal.command = TurnIndicatorsCommand::NO_COMMAND;
+
+  const lanelet::BasicPoint2d current_point{current_pose.position.x, current_pose.position.y};
+  const auto base_search_distance = search_distance + current_vel * search_time;
+
+  std::vector<lanelet::Id> searched_lanelet_ids = {};
+  std::optional<double> arc_length_from_vehicle_front = std::nullopt;
+
+  for (const auto & point : path.points) {
+    const auto & lane_id = point.lane_ids.front();
```

Comment:
`for (const auto & lane_id : point.lane_ids)`

In many cases it contains only one id, but it is better to loop it.